semgrep/semgrep/core_runner.py

```diff
@@ -257,16 +257,14 @@ def _run_core_command(
                 "-max_memory",
                 str(self._max_memory),
                 "-json_time",
+                "-debug",
```

Comment:
@mjambon instead of passing -debug conditionally and outputing debug logs if they exist, I switch to always passing debug logs and conditionally outputing. This assumes that there is no performance hit from using -debug in semgrep-core which I don't imagine there is but thought I would double check with you.

Reply:
I would assume that passing `-debug` can make things significantly slower. Even it's not slow today, it can become slow in the future. I think you should keep it conditional, i.e. reflecting the semgrep command line, requiring either `--debug` or `--debugging-json`, so as to minimize surprises.

Reply:
Makes sense. I returned it to conditionally passing the -debug flag